pango/src/layout.rs

```diff
@@ -0,0 +1,50 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::mem;
+
+use glib::translate::*;
+
+use crate::LayoutLine;
+
+/// The result of [`LayoutLine::x_to_index`].
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct HitPosition {
+    /// The UTF-8 byte offset of the grapheme closest to the position.
+    index: i32,
+    /// The codepoint within the grapheme of the position.
+    ///
+    /// This will always be either `0`, or the number of `char`s (*not bytes!*)
+    /// in the grapheme. This represents whether the user clicked near the start
+    /// of the grapheme or near the end; this is important for things like
+    /// resolving cursor positions.
+    trailing: i32,
+    /// Whether or not the position was within the bounds of the line.
+    ///
+    /// If this is `false`, then `index` and `trailing` will always resolve
+    /// to either the very first or the very last position in the line; this
+    /// behaviour is dependent on the line's resolved writing direction.
+    is_inside: bool,
+}
+
+impl LayoutLine {
+    /// Return the [`HitPosition`] corresponding to the provided `x_pos` in this line.
+    pub fn x_to_index(&self, x_pos: i32) -> HitPosition {
```

Comment:
You forgot to add `#[doc(alias = "pango_layout_line_x_to_index")]`.

Reply:
does that mean I should remove my own docs?

Reply:
https://doc.rust-lang.org/rustdoc/advanced-features.html?highlight=alias#add-aliases-for-an-item-in-documentation-search

But I just realized that your doc will be removed in any case because we generate documentation too. (Documentation is a dark place for gtk-rs...)